docs/developer-guide/joins/partition-data.md

```diff
@@ -35,15 +35,6 @@ correct key and partitioning.
     after the repartition. Otherwise, {{ site.ak }} is likely to interleave messages.
     The use case will determine if these ordering guarantees are acceptable.
 
-ksqlDB requires keys to use the `KAFKA` format. For more information, see
-[Serialization Formats](../serialization.md#serialization-formats). If internally
-repartitioning, ksqlDB uses the correct format. If the data in your {{ site.ak }} topics 
-does not have a suitable key format, see [Key Requirements](../syntax-reference.md#key-requirements).
-
-The KAFKA format doesn't support serializing the column name within the data, so the key column name is 
-not important for joins. The key column type is important: for the join to be valid,
```

Comment:
The section titled "Records Have the Same Key Schema" below covers the key column type requirement, so I think it's clearer to remove this text from here even though this restriction is not outdated (the other restrictions are outdated).